modules/nextflow/src/main/groovy/nextflow/extension/ChannelExtensionProvider.groovy

```diff
@@ -170,6 +187,14 @@ class ChannelExtensionProvider implements ExtensionProvider {
         def result = new HashSet<String>(30)
         def methods = clazz.getDeclaredMethods()
         for( def handle : methods ) {
+            // in a future only annotated methodS will be imported
+            if( handle.isAnnotationPresent(Factory)) {
+                def returnType =handle.getReturnType()
+                if( isWriteChannel(returnType) ) {
```

Comment:
Also here, let's report an error if the return type is wrong and a warning the deprecated case